Left rotate source table for the `quickwit source describe` command

The `quickwit source describe` command  outputs three tables:
- source table
- params table
- checkpoint table

The source table should be [left-rotated](https://github.com/zhiburt/tabled#Rotate) for a slicker display. However, when I tried to do so, I did not manage to get the content of the table properly left-aligned.

Hi, I have experimented with different alignments and formats but the output seems to be very off. For example, without PSQL
```
       +------+------------------+
       | Type |       file       |
+------+------+------------------+
 Source|  ID  | wikipedia-source |
+------+------+------------------+


           +-------+--------------------------------------+
           | Value | "/quickwit/wiki-articles-10000.json" |
+----------+-------+--------------------------------------+
 Parameters|  Key  |               filepath               |
+----------+-------+--------------------------------------+


           +--------------+
           |    Offset    |
+----------+--------------+
 Checkpoint| Partition ID |
+----------+--------------+
```
I would assume in this case we need dynamic tables as mentioned in the docs of the builder module https://docs.rs/tabled/latest/tabled/builder/struct.Builder.html
I think if I experiment a bit more, I might be able to figure out the correct way to display the data. However, I am new to rust and I can't seem to figure out the reason behind this unresolved import when I try to import the builder::Builder module. 

```rs
error[E0432]: unresolved import `tabled::builder`
  --> quickwit-cli/src/lib.rs:32:90
   |
32 | use tabled::{Alignment, Rotate, Header, Modify, Row, Full, Style, Table, Tabled, Indent, builder::Builder};
   |                                                                                          ^^^^^^^ could not find `builder` in `tabled`
   ```
  Would appreciate any help! Thank you!

--- a/quickwit-cli/Cargo.toml
+++ b/quickwit-cli/Cargo.toml
@@ -37,7 +37,7 @@ quickwit-serve = { version = "0.2.1", path = "../quickwit-serve" }
 quickwit-telemetry = { version = "0.2.1", path = "../quickwit-telemetry" }
 quickwit-pushapi = {version="0.2.1", path = "../quickwit-pushapi" }
 quickwit-proto = { version = "0.2.1", path = "../quickwit-proto" }
-tabled = "0.5"
+tabled = "0.6"
 tracing = "0.1.29"
 tracing-subscriber = {version="0.3", features=["time", "std", "env-filter"]}
 tracing-opentelemetry = "0.17"

--- a/quickwit-cli/src/lib.rs
+++ b/quickwit-cli/src/lib.rs
@@ -29,7 +29,7 @@ use quickwit_indexing::check_source_connectivity;
 use quickwit_metastore::quickwit_metastore_uri_resolver;
 use quickwit_storage::{load_file, quickwit_storage_uri_resolver};
 use regex::Regex;
-use tabled::{Alignment, Header, Modify, Row, Style, Table, Tabled};
+use tabled::{Alignment, Header, Modify, Rotate, Rows, Style, Table, Tabled};
 use tracing::info;
 
 pub mod cli;
@@ -110,10 +110,25 @@ pub async fn run_index_checklist(
             ));
         }
     }
-
     run_checklist(checks);
-
     Ok(())
+}
+
+/// Constructs a table for display.
+pub fn make_table<T: Tabled>(
+    header: &str,
+    rows: impl IntoIterator<Item = T>,
+    rotate: bool,
+) -> Table {
+    let mut table = Table::new(rows)
+        .with(Modify::new(Rows::new(1..)).with(Alignment::left()))
+        .with(Style::ascii());
+    if rotate {
+        table = table.with(Rotate::Left)
+    }
+    table
+        .with(Header(header))
+        .with(Modify::new(Rows::single(0)).with(Alignment::center()))
 }
 
 #[cfg(test)]
@@ -142,12 +157,4 @@ mod tests {
         assert!(parse_duration_with_unit("1h30").is_err());
         Ok(())
     }
-}
-
-/// Construct a table for display.
-pub fn make_table<T: Tabled>(header: &str, rows: impl IntoIterator<Item = T>) -> Table {
-    Table::new(rows)
-        .with(Header(header))
-        .with(Modify::new(Row(2..)).with(Alignment::left()))
-        .with(Style::psql())
 }

--- a/quickwit-cli/src/source.rs
+++ b/quickwit-cli/src/source.rs
@@ -295,13 +295,13 @@ where
         source_id: source.source_id.clone(),
         source_type: source.source_type().to_string(),
     }];
-    let source_table = make_table("Source", source_rows);
+    let source_table = make_table("Source", source_rows, true);
 
     let params_rows = flatten_json(source.params())
         .into_iter()
         .map(|(key, value)| ParamsRow { key, value })
         .sorted_by(|left, right| left.key.cmp(&right.key));
-    let params_table = make_table("Parameters", params_rows);
+    let params_table = make_table("Parameters", params_rows, false);
 
     let checkpoint_rows = checkpoint
         .iter()
@@ -310,7 +310,7 @@ where
             offset: position.as_str().to_string(),
         })
         .sorted_by(|left, right| left.partition_id.cmp(&right.partition_id));
-    let checkpoint_table = make_table("Checkpoint", checkpoint_rows);
+    let checkpoint_table = make_table("Checkpoint", checkpoint_rows, false);
     Ok((source_table, params_table, checkpoint_table))
 }
 
@@ -331,30 +331,30 @@ where I: IntoIterator<Item = SourceConfig> {
             source_id: source.source_id,
         })
         .sorted_by(|left, right| left.source_id.cmp(&right.source_id));
-    make_table("Sources", rows)
+    make_table("Sources", rows, false)
 }
 
 #[derive(Tabled)]
 struct SourceRow {
-    #[header("ID")]
-    source_id: String,
-    #[header("Type")]
+    #[tabled(rename = "Type")]
     source_type: String,
+    #[tabled(rename = "ID")]
+    source_id: String,
 }
 
 #[derive(Tabled)]
 struct ParamsRow {
-    #[header("Key")]
+    #[tabled(rename = "Key")]
     key: String,
-    #[header("Value")]
+    #[tabled(rename = "Value")]
     value: Value,
 }
 
 #[derive(Tabled)]
 struct CheckpointRow {
-    #[header("Partition ID")]
+    #[tabled(rename = "Partition ID")]
     partition_id: String,
-    #[header("Offset")]
+    #[tabled(rename = "Offset")]
     offset: String,
 }
 
@@ -558,7 +558,7 @@ mod tests {
         )
         .is_err());
 
-        let checkpoint: SourceCheckpoint = vec![("shard-000", ""), ("shard-001", "42")]
+        let checkpoint: SourceCheckpoint = vec![("shard-000", ""), ("shard-001", "1234567890")]
             .into_iter()
             .map(|(partition_id, offset)| (PartitionId::from(partition_id), Position::from(offset)))
             .collect();
@@ -581,22 +581,22 @@ mod tests {
             },
             CheckpointRow {
                 partition_id: "shard-001".to_string(),
-                offset: "42".to_string(),
+                offset: "1234567890".to_string(),
             },
         ];
         let (source_table, params_table, checkpoint_table) =
             make_describe_source_tables(checkpoint, sources, "foo-source").unwrap();
         assert_eq!(
             source_table.to_string(),
-            make_table("Source", expected_source).to_string()
+            make_table("Source", expected_source, true).to_string()
         );
         assert_eq!(
             params_table.to_string(),
-            make_table("Parameters", expected_params).to_string()
+            make_table("Parameters", expected_params, false).to_string()
         );
         assert_eq!(
             checkpoint_table.to_string(),
-            make_table("Checkpoint", expected_checkpoint).to_string()
+            make_table("Checkpoint", expected_checkpoint, false).to_string()
         );
     }
 
@@ -645,7 +645,7 @@ mod tests {
         ];
         assert_eq!(
             make_list_sources_table(sources).to_string(),
-            make_table("Sources", expected_sources).to_string()
+            make_table("Sources", expected_sources, false).to_string()
         );
     }
 }

--- a/quickwit-cli/src/split.rs
+++ b/quickwit-cli/src/split.rs
@@ -430,7 +430,7 @@ fn make_list_splits_table(splits: Vec<Split>) -> Table {
                 id: split.split_metadata.split_id,
                 num_docs: split.split_metadata.num_docs,
                 size_mega_bytes: split.split_metadata.original_size_in_bytes / 1_000_000,
-                create_at: OffsetDateTime::from_unix_timestamp(
+                created_at: OffsetDateTime::from_unix_timestamp(
                     split.split_metadata.create_timestamp,
                 )
                 .expect("could not create OffsetDateTime from timestamp"),
@@ -440,7 +440,7 @@ fn make_list_splits_table(splits: Vec<Split>) -> Table {
             }
         })
         .sorted_by(|left, right| left.id.cmp(&right.id));
-    make_table("Splits", rows)
+    make_table("Splits", rows, false)
 }
 
 fn split_state_from_input_str(input: &str) -> anyhow::Result<SplitState> {
@@ -457,17 +457,17 @@ fn split_state_from_input_str(input: &str) -> anyhow::Result<SplitState> {
 
 #[derive(Tabled)]
 struct SplitRow {
-    #[header("Id")]
+    #[tabled(rename = "ID")]
     id: String,
-    #[header("Num Docs")]
+    #[tabled(rename = "Num docs")]
     num_docs: usize,
-    #[header("Size (MB)")]
+    #[tabled(rename = "Size (MB)")]
     size_mega_bytes: u64,
-    #[header("Created At")]
-    create_at: OffsetDateTime,
-    #[header("Updated At")]
+    #[tabled(rename = "Created at")]
+    created_at: OffsetDateTime,
+    #[tabled(rename = "Updated at")]
     updated_at: OffsetDateTime,
-    #[header("Time Range")]
+    #[tabled(rename = "Time range")]
     time_range: String,
 }
 

